### PR TITLE
Feature support oauth2 for node client

### PR DESCRIPTION
**Pull request by tuteng**
* The cpp client already supports oauth2 authentication, and some users want to support this feature in the node js client as well, so enable it in the node js client

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,9204 +1,8 @@
 {
   "name": "pulsar-client",
   "version": "1.6.0-rc.0",
-  "lockfileVersion": 2,
```
Comment (jiazhai) [inline package-lock.json]: Seems we removed a lot of code, what were these codes used for?

**Review by jiazhai — APPROVED** [review]

**Review by gaoran10 — APPROVED** [review]
Lgtm

File: package-lock.json (hunk 1 above)
Comment (tuteng) [reply package-lock.json]: Thanks, revert this change

**Comment by hassanzia32** [comment]
Is there a timeline for when the 1.6.0 version is scheduled to be released?

**Pull request merged**